tests/backup/backup_helper.go

```diff
@@ -6545,3 +6564,72 @@ func SuspendAndDeleteSchedule(backupScheduleName string, schedulePolicyName stri
 	}
 	return nil
 }
+
+// ValidateCustomResourceRestores validates restore taken of custom resource
+func ValidateCustomResourceRestores(ctx context.Context, orgID string, resourceTypesFilter []string, restoreContextMap map[string][]*scheduler.Context, clusterName string) error {
+	if clusterName == "source-cluster" {
+		err := SetSourceKubeConfig()
+		if err != nil {
+			return err
+		}
+	} else if clusterName == "destination-cluster" {
+		err := SetDestinationKubeConfig()
+		if err != nil {
+			return err
+		}
+	} else {
+		return fmt.Errorf("cluster name provided is not correct %s", clusterName)
+	}
+	errChan := make(chan error, 100)
+	var errList []error
+	var wg sync.WaitGroup
+	for restoreName, contexts := range restoreContextMap {
+		restoreInspectRequest := &api.RestoreInspectRequest{
+			Name:  restoreName,
+			OrgId: orgID,
+		}
+		response, err := Inst().Backup.InspectRestore(ctx, restoreInspectRequest)
+		if err != nil {
+			return err
+		}
+		restoreObj := response.GetRestore()
+		nsMapping := restoreObj.NamespaceMapping
+		scMapping := restoreObj.StorageClassMapping
+		log.Infof("Namespace mapping is %s, storage class mapping is %s for restore %s", nsMapping, scMapping, restoreName)
+		wg.Add(1)
+		go func(restoreName string, nsMapping map[string]string, scMapping map[string]string) {
+			defer GinkgoRecover()
+			defer wg.Done()
+			log.InfoD("Validating restore [%s]", restoreName)
+			var expectedRestoredAppContextList []*scheduler.Context
+			for _, context := range contexts {
+				expectedRestoredAppContext, err := CloneAppContextAndTransformWithMappings(context, nsMapping, scMapping, true)
+				if err != nil {
+					errChan <- err
+					return
+				}
+				expectedRestoredAppContextList = append(expectedRestoredAppContextList, expectedRestoredAppContext)
+			}
+			err := ValidateRestore(ctx, restoreName, orgID, expectedRestoredAppContextList, resourceTypesFilter)
```

Comment:
Are we logging anywhere about the resource type we are validating here?

Reply:
done